services/app-api/src/postgres/contractAndRates/proto_to_db_RateRevisions.ts

```diff
@@ -0,0 +1,179 @@
+                    create: {
+                        updatedAt: formData.updatedAt,
+                        updatedByID: user.id,
+                        updatedReason:
+                            revision.submittedReason ??
+                            'Migrated from previous system',
+                    },
+                }
+            } else {
+                console.warn(
+                    `User with email ${revision.submittedBy} does not exist. Skipping submitInfo creation.`
+                )
+            }
+        }
+
+        // add the actuary contacts
+        if (rateInfo.actuaryContacts) {
+            dataToCopy.certifyingActuaryContacts = {
+                create: rateInfo.actuaryContacts,
+            }
+        }
+
+        if (formData.addtlActuaryContacts) {
+            dataToCopy.addtlActuaryContacts = {
+                create: formData.addtlActuaryContacts,
+            }
+        }
+        // handle rate documents
+        let rateDocPos = 0
+        const rateDocsArray = []
+        for (const rateDoc of rateInfo.rateDocuments) {
+            const rateDocument: Prisma.RateDocumentCreateWithoutRateRevisionInput =
+                {
+                    createdAt: revision.createdAt,
+                    updatedAt: new Date(),
+                    name: rateDoc.name,
+                    s3URL: rateDoc.s3URL,
+                    sha256: rateDoc.sha256,
+                    position: rateDocPos,
+                }
+            rateDocsArray.push(rateDocument)
+            rateDocPos++
+        }
+        dataToCopy.rateDocuments = {
+            create: rateDocsArray,
+        }
+
+        // handle rate revision documents
+        let rateRevDocPos = 0
+        const rateRevDocsArray = []
+        for (const supportRateDoc of rateInfo.supportingDocuments) {
+            const rateSupportDocument: Prisma.RateSupportingDocumentCreateWithoutRateRevisionInput =
+                {
+                    createdAt: revision.createdAt,
+                    updatedAt: new Date(),
+                    name: supportRateDoc.name,
+                    s3URL: supportRateDoc.s3URL,
+                    sha256: supportRateDoc.sha256,
+                    position: rateRevDocPos,
+                }
+            rateRevDocsArray.push(rateSupportDocument)
+            rateRevDocPos++
+        }
+        dataToCopy.supportingDocuments = {
+            create: rateRevDocsArray,
+        }
+
+        // rate revisions on contract revisions join table
+        dataToCopy.contractRevisions = {
+            create: {
+                contractRevisionID: contractRevision.id,
+                validAfter: new Date(),
+            },
+        }
+
+        rateRevisionData.push(dataToCopy)
+    }
+
+    const insertRateData: Prisma.RateTableCreateInput = {
+        state: {
+            connect: {
+                stateCode: stateCode,
+            },
+        },
+        stateNumber: stateNumber,
```

Comment:
This will have to be a new number. Rates have their own state number count in the `State` table `latestStateRateCertNumber`. We should add code to increment `latestStateRateCertNumber` every time a new rate is inserted or have a total to update at once if we are inside a transaction.